src/dividat-driver/firmware/main.go

```diff
@@ -90,7 +97,7 @@ func Update(parentCtx context.Context, image io.Reader, deviceSerial *string, co
 		dfuHost = *configuredAddr
 	} else {
 		ctx, cancel := context.WithTimeout(parentCtx, 60*time.Second)
-		discoveredAddr, err := discover("_sensoUpdate._udp", deviceSerial, ctx)
+		discoveredAddr, err := discover("_sensoUpdate._udp", deviceSerial, ctx, onProgress)
```

Comment:
The address could change here, right? Should we compare serial numbers to see if we're rediscovering the same Senso as the one we meant to update?

Reply:
I think it would be best to start the update process by a device serial passed from the client to the driver, as the address indeed might change.

It would be good to also allow passing a fixed address, though, which Play could do if a connection is configured in this way instead of via serial. (I think it would be OK to only offer the serial option initially.)

Reply:
We can deduce the serial for fixed address configurations from the records returned in the discovery response.